src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -796,6 +804,38 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
         return [$objtype, $objval, $offset];
     }
 
+    /**
+     * Get value of an object header's section
+     *
+     * @param string      $key     header's section name
+     * @param string      $type    type of the section (i.e. 'numeric', '/', '<<', etc.)
+     * @param string|null $default default value for header's section
+     *
+     * @return string|array|null value of obj header's section, or default value if none found, or its type doesn't match $type param
+     */
+    private function getHeaderValue(?array $headerDic, string $key, string $type, $default = '')
+    {
+        if (!\is_array($headerDic)) {
+            return $default;
+        }
+
+        foreach ($headerDic as $i => $val) {
+            if (
+                \is_array($val)
+                && 3 == \count($val)
+                && '/' == $val[0]
+                && $val[1] == $key
+                && isset($headerDic[$i + 1])
+            ) {
+                return \is_array($headerDic[$i + 1]) && 1 < \count($headerDic[$i + 1]) && $type == $headerDic[$i + 1][0]
+                    ? $headerDic[$i + 1][1]
+                    : $default;
+            }
```

Comment:
This is a utility function, that provides access to object header's fields (obj << YYY >> part ), like Header::get('...')->getContent() do.
The only difference -- it can be used during the parsing process, when no Smalot\PdfParser\Header objects are created yet.

It recieves dictionary of header fields, as it is returned by RawDataParser::getRawObject,
iterates over it, searching for section of type '/' whith requested key.
If such a section is found, it tries to receive it's value (next object in dictionary), returning it, if it matches requested type, or default value otherwise.

This function can simplify index arithmetic here: 
https://github.com/smalot/pdfparser/blob/65bdad55246ec230a055fbda501c6fed592c1942/src/Smalot/PdfParser/RawData/RawDataParser.php#L262
It's just $valid_crs = 'XRef' == $this->getHeaderValue($sdik, 'Type', '/');
and several ifs below this line;

Here:
https://github.com/smalot/pdfparser/blob/65bdad55246ec230a055fbda501c6fed592c1942/src/Smalot/PdfParser/RawData/RawDataParser.php#L100
And there:
https://github.com/smalot/pdfparser/blob/65bdad55246ec230a055fbda501c6fed592c1942/src/Smalot/PdfParser/RawData/RawDataParser.php#L107

Answering your questions:

1. I don't see any more edge cases, we have to test, since i'm testing correctness of inner structure of an object, returned by RawDataParser::getRawObject (those is_array, count() >= X). Even this may be a bit excessive, but let it be :)
 
2. It's just a separation of responsibility, nothing more. if ensures, that desired field is found and it has some value, ternary operator ensures that value object is correct and has desired type.
It could be reduced to one if, if you insist.